libs/pageserver_api/src/models.rs

```diff
@@ -19,6 +19,22 @@ pub enum TenantState {
     Broken,
 }
 
+/// A state of a timeline in pageserver's memory.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+pub enum TimelineState {
+    /// Timeline is fully operational, its background jobs are running.
+    Active,
+    /// A timeline is recognized by pageserver, but not yet ready to operate.
+    /// The status indicates, that the timeline could eventually go back to Active automatically:
+    /// for example, if the owning tenant goes back to Active again.
+    Suspended,
+    /// A timeline is recognized by pageserver, but not yet ready to operate and not allowed to
+    /// automatically become Active after certain events: only a management call can change this status.
+    Paused,
+    /// A timeline is recognized by the pageserver, but no longer used for any operations, as failed to get activated.
+    Broken,
+}
```

Comment:
I'm a little confused on which state a timeline starts in, and what transitions are allowed. Reading the comment on Suspended, I thought that would be the initial state. But what's this about "going back to Active"? Back from where?

Reply:
'Paused' suggests that you might resume it later. That's not how it's used: when a timeline goes into Paused state, we are deleting or detaching it. It's not coming back. Let's call it 'Stopping'.

Reply:
Also, Suspended and Paused sound like the same thing. If I understand correctly, 'Suspended' means something like 'Starting': it's not open for queries yet, but will be soon.

Reply:
Putting all that together, I suggest:


```suggestion
    /// Timeline is recognized by pageserver, but not yet ready to operate.
    Starting,
    /// Timeline is fully operational, its background jobs are running.
    Active,
    /// Timeline is recognized by pageserver, but we are about to delete or detach it. If the timeline
    /// was Active previously (we could also go directly from Starting to Stopping), you may continue
    /// to operate on it as long as you are associated with the timeline in `task_mgr.rs`. You should
    /// stop as quickly as possible, though, to allow the timeline shutdown to finish.  
    Stopping,
    /// Timeline is recognized by the pageserver, but something went wrong while starting it up.
    Broken,
}
```

Reply:
Oh, also list in a comment the possible state transitions. In my suggestion above, the idea is that the state always moves forward. So it's Starting -> Active -> Stopping.